Mathlib/Data/Finset/Basic.lean

```diff
@@ -3850,6 +3859,36 @@ def sigmaEquivOptionOfInhabited (α : Type u) [Inhabited α] [DecidableEq α] :
           · simp }⟩
 #align equiv.sigma_equiv_option_of_inhabited Equiv.sigmaEquivOptionOfInhabited
 
+variable [DecidableEq α] {s t : Finset α}
+
+/-- `s ∪ t` (using finset union) is equivalent to `s ∪ t` (using set union) -/
+@[simps!]
+def finsetUnion (s t : Finset α) :
+    (s ∪ t : Set α) ≃ ((s ∪ t : Finset α) : Set α) :=
+  Equiv.Set.ofEq <| coe_union _ _ |>.symm
+
+/-- The disjoint union of finsets is a sum -/
+def Finset.union (s t : Finset α) (h : Disjoint s t) :
+    s ⊕ t ≃ (s ∪ t : Finset α) :=
+  Equiv.Set.union (disjoint_coe.mpr h).le_bot |>.symm.trans <| Equiv.finsetUnion s t
```

Comment:
I think it would be worth having a `disjUnion` version of this, since then you could drop the `[DecidableEq α]` assumption.

Reply:
Then we would need to add `DecidablePred (· ∈ s)` to make `Equiv.Set.union` work. I don't see how that's better.

Reply:
Oh good point, you need to be able to decide which half of the disjoint union an element is in to produce the sum type.